core/trino-main/src/main/java/io/trino/operator/output/RowPositionsAppender.java

```diff
@@ -100,20 +110,29 @@ public void append(IntArrayList positions, Block block)
     public void appendRle(Block value, int rlePositionCount)
     {
         ensureCapacity(rlePositionCount);
-        AbstractRowBlock sourceRowBlock = (AbstractRowBlock) value;
-        if (sourceRowBlock.isNull(0)) {
+        if (value instanceof AbstractRowBlock sourceRowBlock) {
```

Comment:
nit: I think there might be other places where we assume `AbstractRowBlock` always